pandas/tests/reductions/test_reductions.py

```diff
@@ -66,60 +66,64 @@ def test_ops(self, opname, obj):
-                if klass is Series:
-                    assert np.isnan(result)
-                else:
-                    assert result == -1
-
-                obj = klass([pd.NaT, datetime(2011, 11, 1), pd.NaT])
-                # check DatetimeIndex non-monotonic path
-                assert getattr(obj, opname)(), datetime(2011, 11, 1)
-                assert getattr(obj, opname)(skipna=False) is pd.NaT
-
-                assert getattr(obj, arg_op)() == 1
-                result = getattr(obj, arg_op)(skipna=False)
-                if klass is Series:
-                    assert np.isnan(result)
-                else:
-                    assert result == -1
-
-                for dtype in ["M8[ns]", "datetime64[ns, UTC]"]:
-                    # cases with empty Series/DatetimeIndex
-                    obj = klass([], dtype=dtype)
-
-                    assert getattr(obj, opname)() is pd.NaT
-                    assert getattr(obj, opname)(skipna=False) is pd.NaT
-
-                    with pytest.raises(ValueError, match="empty sequence"):
-                        getattr(obj, arg_op)()
-                    with pytest.raises(ValueError, match="empty sequence"):
-                        getattr(obj, arg_op)(skipna=False)
-
-        # argmin/max
+        klass = index_or_series
+        arg_op = "arg" + opname if klass is Index else "idx" + opname
+
+        obj = klass([np.nan, 2.0])
+        assert getattr(obj, opname)() == 2.0
+
+        obj = klass([np.nan])
+        assert pd.isna(getattr(obj, opname)())
+        assert pd.isna(getattr(obj, opname)(skipna=False))
+
+        obj = klass([], dtype=object)
+        assert pd.isna(getattr(obj, opname)())
+        assert pd.isna(getattr(obj, opname)(skipna=False))
+
+        obj = klass([pd.NaT, datetime(2011, 11, 1)])
+        # check DatetimeIndex monotonic path
+        assert getattr(obj, opname)() == datetime(2011, 11, 1)
+        assert getattr(obj, opname)(skipna=False) is pd.NaT
+
+        assert getattr(obj, arg_op)() == 1
+        result = getattr(obj, arg_op)(skipna=False)
+        if klass is Series:
+            assert np.isnan(result)
+        else:
+            assert result == -1
+
+        obj = klass([pd.NaT, datetime(2011, 11, 1), pd.NaT])
+        # check DatetimeIndex non-monotonic path
+        assert getattr(obj, opname)(), datetime(2011, 11, 1)
+        assert getattr(obj, opname)(skipna=False) is pd.NaT
+
+        assert getattr(obj, arg_op)() == 1
+        result = getattr(obj, arg_op)(skipna=False)
+        if klass is Series:
+            assert np.isnan(result)
+        else:
+            assert result == -1
+
+    @pytest.mark.parametrize("opname", ["max", "min"])
+    @pytest.mark.parametrize("dtype", ["M8[ns]", "datetime64[ns, UTC]"])
+    def test_nanops_empty_object(self, opname, index_or_series, dtype):
+        klass = index_or_series
+        arg_op = "arg" + opname if klass is Index else "idx" + opname
```

Comment:
why is this needed?

Reply:
I broke this into its own test since it allows for parametrizing over these dtypes without repeating all of `test_nanops` (which was / is pretty big), and this was conceptually a bit more specific than just "testing nanops"

Reply:
i get that, but why is arg_op specifically needed? what is its purpose

Reply:
Oh, I think the name of the method being tested changes if it's a Series or Index (idxmin or argmin) so it has to be created dynamically in the test?